hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CustomFeesValidator.java

```diff
@@ -245,4 +250,29 @@ private void validateRoyaltyFeeForCreation(
             }
         }
     }
+
+    private void validateFractionalFeeForFeeScheduleUpdate(
+            @NonNull final Token token,
+            @NonNull final ReadableTokenRelationStore tokenRelationStore,
+            @NonNull final AccountID collectorId,
+            @NonNull final CustomFee fee) {
+        final var tokenType = token.tokenType();
+        // fractional fee can be only applied to fungible common tokens
+        validateTrue(isFungibleCommon(tokenType), CUSTOM_FRACTIONAL_FEE_ONLY_ALLOWED_FOR_FUNGIBLE_COMMON);
+        final var tokenId = token.tokenId();
+        final var relation = tokenRelationStore.get(collectorId, tokenId);
+        validateTrue(relation != null, TOKEN_NOT_ASSOCIATED_TO_FEE_COLLECTOR);
+
+        final var fractionalFee = fee.fractionalFee();
+        final var numerator = fractionalFee.fractionalAmount().numerator();
+        final var denominator = fractionalFee.fractionalAmount().denominator();
+        final var minimumUnitsToCollect = fractionalFee.minimumAmount();
+        final var nominalMax = fractionalFee.maximumAmount();
+        final var maximumUnitsToCollect = nominalMax == 0 ? Long.MAX_VALUE : nominalMax;
+
+        validateTrue(denominator != 0, FRACTION_DIVIDES_BY_ZERO);
+        validateTrue(numerator > 0 && denominator > 0, CUSTOM_FEE_MUST_BE_POSITIVE);
+        validateTrue(minimumUnitsToCollect >= 0 && maximumUnitsToCollect >= 0, CUSTOM_FEE_MUST_BE_POSITIVE);
+        validateTrue(maximumUnitsToCollect >= minimumUnitsToCollect, FRACTIONAL_FEE_MAX_AMOUNT_LESS_THAN_MIN_AMOUNT);
```

Comment:
I think there are some functions added for common logic in this PR https://github.com/hashgraph/hedera-services/pull/8672. We should probably re-use the same functions to avoid duplicating code. Can be done after the PR is merged also.

Reply:
Also some logic is missing for Royalty Fee as specified in the above PR. Should fix those as well.

Reply:
Yeah, I agree with you @Neeharika-Sompalli 
I think since #8672 was merged, I can update this branch by re-using logic from it

Reply:
Sounds good. Thanks @thenswan

Reply:
@Neeharika-Sompalli updated PR